AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/hierarchical-state-machine-swift/StateMachine.swift

```diff
@@ -73,70 +67,41 @@ class StateMachine<
     typealias OnSubscribedCallback = () -> Void
 
     private let environment: EnvironmentType
-    private let executor: EffectExecutor
     private let resolver: AnyResolver<StateType>
 
-    /// Backing queue for both consistencyQueue and notificationQueue
-    private let concurrentQueue: DispatchQueue
-
-    /// Manages consistency of internal state machine state
-    let operationQueue: OperationQueue
-
-    private var currentState: StateType
+    var currentState: StateType
```

Comment:
Should make this a `private(set) var` right? We don't want anything except the state machine resolver logic to update the state.